src/poetry/utils/dependency_specification.py

```diff
@@ -167,7 +172,8 @@ def pep508_to_dependency_specification(requirement: str) -> DependencySpec | Non
 
     with contextlib.suppress(ValueError):
         dependency = Dependency.create_from_pep_508(requirement)
-        specification = dependency_to_specification(dependency)
+        specification: DependencySpec = {}
```

Comment:
Can the type not be declared on the same line, or am I missing something?

Reply:
Since i am using generics for the factory, mypy complains. This is more to appease the mighty mypy.